dspace-api/src/main/java/org/dspace/content/authority/InputFormSelfRegisterWrapperAuthority.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content.authority;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.dspace.app.util.DCInputsReader;
+import org.dspace.app.util.DCInputsReaderException;
+import org.dspace.content.Collection;
+import org.dspace.core.Utils;
+
+/**
+ * This authority is registered automatically by the ChoiceAuthorityService for
+ * all the metadata that use a value-pair or a vocabulary in the submission-form.xml
```

Comment:
Isn't this the same purpose as `DCInputAuthority ` (see https://github.com/DSpace/DSpace/blob/master/dspace/config/dspace.cfg#L1446)

Would it be possible to merge these two classes or to reuse `DCInputAuthority`?

Reply:
This is the wrapper to enhanced submission-forms.xml to have different authority registered for a metadata. Think the dc.type and the possibility to show different valuepairs depends on the configuration form (e.g. Thesis dc.type can be different for generic Publication). This class wrap both DCInputAuthority then DSpaceControlledVocabulary .

Reply:
Wouldn't it make more sense to not create`InputFormSelfRegisterWrapperAuthority` and adjust `DCInputAuthority` and `DSpaceControlledVocabulary` to the new requirements? Otherwise I don't see a reason to keep `DCInputAuthority` and `DSpaceControlledVocabulary` if they are not usable without `InputFormSelfRegisterWrapperAuthority`.